content/faq.md

```diff
@@ -55,51 +76,82 @@ It’s also worth noting that, while there are almost 8B people on Earth, there
+OS](#what-is-arvo) itself. But the first challenge is getting Urbit OS to be
+completely secure.
 
 ### Why is Hoon so weird? {#why-hoon}
 
-There are things about Urbit that are weird by design, and some that are weird because we haven’t gotten around to cleaning them up. Hoon, for the most part, is weird by design, but there are a few things about it we’d like to clean up.
+There are things about Urbit that are weird by design, and some that are weird because we haven’t gotten around to cleaning them up. [Hoon](@/docs/glossary/hoon.md), for the most part, is weird by design, but there are a few things about it we’d like to clean up.
 
 The runes are, once you get the hang of them, pretty nice. Your mileage may vary, of course, but many talented engineers have spent many, many hours writing Hoon. It’s a simple, practical language. But the learning curve can be a bit high.
 
 The standard library and naming is something we’ve discussed changing, and it could happen. It’s definitely up for debate whether or not the four-letter naming convention works well. The vanes (kernel modules) have largely moved away from this convention.
 
-The right way to think about Hoon is as the ‘C of Urbit’. It’s a bare bones language that gets you pretty close to Nock. Eventually, we’ll probably implement more expressive languages that compile to Urbit. But for now, satisfy your curiosity and [sign up for Hoon School](/hoonschool/).
-
-### What is the point of Nock? {#why-nock}
+The right way to think about Hoon is as the ‘C of Urbit’. It’s a bare bones
+language that gets you pretty close to Nock. Eventually, we’ll probably
+implement more expressive languages that compile to Urbit. But for now, satisfy
+your curiosity and [sign up for Hooniversity](https://hooniversity.org/), a
+community led Hoon educational organization.
 
-The primary technical reason we can’t run our own computers already is that they don’t all compute the same way. Nock solves this problem. It provides a single computing foundation for the whole network. It’s also deterministic — which is an important feature for anything you’re going to put sensitive data into.
+Another option is to treat [Urbit as an API](@/using/integrating-api.md), which
+allows one to use common languages such as Python and Javascript to build
+userspace programs. For now this option is not as powerful as Hoon, but
+there is still a lot that a developer without the time or inclination to learn
+Hoon can do.
 
-It’s true that Nock, without jets, is slow. With the ability to call out to a native function, it’s not so bad. We’ve been working on a new Nock interpreter for Grall Truffle that’s about 10x faster than the current one — so we’re not all that worried about performance for the time being.
+### What is the point of Nock? {#why-nock}
 
-### Can I really host my Urbit with someone else? {#why-hosting}
+The primary technical reason we can’t run our own computers already is that they
```

Comment:
This assertion seems very weird. We can't run our own computers in the way that urbit is intended to solve not because they aren't all x86 asm. We can't run them because corporations have a vested interest in centralization of services and build walled gardens to prevent you from interacting with your data there outside of the ways they deem acceptable.

Reply:
Yeah this didn't make any sense. I rewrote it to focus on something else, and I think its an improvement but could still do better. I'll think about it some more. https://github.com/urbit/urbit.org/pull/538/commits/a68bb0ddf99d8add23f4556d3e9779e8ca05df31